.github/workflows/presubmit.yml

```diff
@@ -171,42 +179,33 @@ jobs:
         run: |
           set -x
           echo "DIST_DIR=$HOME/dist" >> $GITHUB_ENV
+          
           # use latest NDK because the default NDK on github is older.
           # TODO b/216535050: Implement task to install the exact AndroidX ndk version.
           echo "ANDROID_NDK_ROOT=$ANDROID_NDK_LATEST_HOME" >> $GITHUB_ENV
-      # gradle action loads the dependencies cache only on the first run based on arguments.
-      # to control it, we explicitly invoke it once which makes it load the dependencies cache with the parameters
-      # we control
-      - name: "warm up gradle"
-        id: warm-up-gradle-cache
-        uses: gradle/gradle-command-action@v1
-        env:
-          JAVA_HOME: ${{ steps.setup-java.outputs.path }}
-          JAVA_TOOLS_JAR: ${{ steps.setup-tools-jar.outputs.toolsJar }}
+
+      - name: "Setup Gradle"
+        uses: gradle/gradle-build-action@v2
         with:
-          arguments: projects -PandroidXUnusedParameter=${{ env.project-root }} # add project name to cache key
-          build-root-directory: ${{ env.project-root }}
-          configuration-cache-enabled: false
-          dependencies-cache-enabled: true
-          dependencies-cache-key: |
-            **/libs.versions.toml
-          dependencies-cache-exact: false
-          gradle-executable: ${{ env.project-root }}/gradlew
-          wrapper-directory: ${{ env.project-root }}/gradle/wrapper
-          distributions-cache-enabled: true
-      - name: "./gradlew buildOnServer zipTestConfigsWithApks test"
-        uses: gradle/gradle-command-action@v1
+          # Only save Gradle User Home state for builds on the 'androidx-main' branch.
+          # Builds on other branches will only read existing entries from the cache.
+          cache-read-only: ${{ github.ref != 'refs/heads/androidx-main' }}
+
+          # Don't reuse cache entries from any other Job.
+          gradle-home-cache-strict-match: true
+
+          # Limit the size of the cache entry.
+          # These directories contain instrumented/transformed dependency jars which can be reconstructed relatively quickly.
+          gradle-home-cache-excludes: | 
+            caches/jars-9
```

Comment:
Do we need something like jars-* instead of 9? (Idk if * is supported but 9 seems very random and could break).

Reply:
Unfortunately this parameter doesn't (yet) support wildcards. I'll add an issue to `gradle-build-action` for this.

The `jars-9` is tied to a set of Gradle versions, and will only be incremented when the instrumentation algorithm changes in Gradle. If you update to a new Gradle version with a new `jars-` version, then the cache entry will contain these files. Things will keep working, but we might start to see cache entry eviction based on hitting the 10Gb limit.

In the end, I'm not certain that these exclusions are necessary, but in my testing they allowed a bit of headroom for growth in entry size without leading to premature eviction. I'd like to get this in and then start to monitor the real-world behaviour, tweaking as necessary.

Reply:
The `gradle-build-action` makes it easy to monitor the size of entries being uploaded: eg https://github.com/bigdaz/androidx/runs/4974841719?check_suite_focus=true#step:17:221. So this will allow us to monitor how well the Gradle User Home caching is performing.
I'm sure further tweaks will be required.